src/mscorlib/src/System/Diagnostics/Eventing/XplatEventLogger.cs

```diff
@@ -78,16 +78,17 @@ private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnly
             if (payloadName.Count == 0 || payload.Count == 0)
                 return String.Empty;
 
-            Contract.Assert(payloadName.Count == payload.Count);
+            int eventDataCount = payloadName.Count;
+
             if(payloadName.Count != payload.Count)
             {
-                return string.Empty;
+               eventDataCount = Math.Min(payloadName.Count, payload.Count);
             }
             
             var sb = StringBuilderCache.Acquire();
 
             sb.Append('{');
-            for (int i = 0; i < payloadName.Count; i++)
+            for (int i = 0; i < eventDataCount; i++)
             {
                 var fieldstr = payloadName[i].ToString();
```

Comment:
If you have a partial payload does the partial payload always start with the first arg and end at N-K, K being some intermediate payload field?  If it is possible to pick and choose which payload fields to specify, then it looks like this fix won't work, as the payload names are matched to payload fields by index into the array.